Products/Zuul/catalog/indexable.py

```diff
@@ -322,6 +331,50 @@ def idx_deviceOrganizers(self):
                 organizers.append( "/".join(sp[:-2]) )
         return organizers
 
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="tagNumber")
+    def idx_tagNumber(self):
+        return self.hw.tag
+ 
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="pythonClass")
+    def idx_pythonClass(self):
+        return self.__class__.__module__
+
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="priority")
+    def idx_priority(self):
+        return self.priority
+
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="collector")
+    def idx_collector(self):
+        return self.getPerformanceServerName()
+
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="osModel")
+    def idx_osModel(self):
+        if self.os.productClass():
+            return self.os.productClass().id
+
+        return None
+
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="osManufacturer")
+    def idx_osManufacturer(self):
+        if self.os.productClass():
+            return self.os.productClass().manufacturer().id
+
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="hwModel")
+    def idx_hwModel(self):
+        if self.hw.productClass():
+            return self.hw.productClass().id
+
+        return None
+
+    @indexed(StringFieldType(indexed=False, stored=True), attr_query_name="hwManufacturer")
+    def idx_hwManufacturer(self):
+        if self.hw.productClass():
+            return self.hw.productClass().manufacturer().id
```

Comment:
Same as above

Reply:
In that case self.os.productClass() is also None.